opentimestamps/bitcoin.py

```diff
@@ -32,7 +32,7 @@ def __make_btc_block_merkle_tree(blk_txids):
     return digests[0]
 
 
-def make_timestamp_from_block(digest, block, blockheight, *, max_tx_size=1000):
+def make_timestamp_from_block(digest, block, blockheight, *, max_tx_size=1000, serde_txs=None):
```

Comment:
What's the rational for allowing `serde_txs` to be provided as an argument?

Reply:
during high fee period in last december or so I noticed high cpu usage by the calendar process
there was a lot of prior transaction in queue in the calendar and investigating I noticed the process was taking a lot of time to process this cycle https://github.com/opentimestamps/opentimestamps-server/blob/master/otsserver/stamper.py#L246
the make_timestamp_from_block is serializing all the tx for every prior transaction and it turns out it's not a cost free operation. To confirm this try to run the provided `test_bench` function with or without that array precomputed to see the differences

Reply:
If you are referring to be a named argument, it's just for being backward compatible to any other caller (lite in tests for example)

Reply:
Oh, I mean, in addition to precalculating it (a good thing!) why also allow it to be provided as an argument? Seems like a premature optimization that unnecessarily changes the API; we're already fixing 99% of the performance problem by simply precalculating it at the beginning of the function.

Reply:
Yeah, I think it'd be better to not allow it to be provided by the caller. Not a major performance improvement, and quite possible to screw up.

Reply:
the saving are because it's precomputed out of the loop https://github.com/opentimestamps/opentimestamps-server/blob/master/otsserver/stamper.py#L246, not because it's computed inside the function